packages/codecs-core/src/__typetests__/combine-codec-typetest.ts

```diff
@@ -0,0 +1,45 @@
+import {
+    Codec,
+    Decoder,
+    Encoder,
+    FixedSizeCodec,
+    FixedSizeDecoder,
+    FixedSizeEncoder,
+    VariableSizeCodec,
+    VariableSizeDecoder,
+    VariableSizeEncoder,
+} from '../codec';
+import { combineCodec } from '../combine-codec';
+
+{
+    // [combineCodec]: It keeps track of the fixed or variable size types.
+    combineCodec({} as FixedSizeEncoder<string>, {} as FixedSizeDecoder<string>) satisfies FixedSizeCodec<string>;
+    combineCodec(
+        {} as VariableSizeEncoder<string>,
+        {} as VariableSizeDecoder<string>
+    ) satisfies VariableSizeCodec<string>;
+    combineCodec({} as Encoder<string>, {} as Decoder<string>) satisfies Codec<string>;
```

Comment:
Is it valid to combine a `FixedSizeEncode` and a `VariableSizeDecoder` or vice versa? It looks like if you did that it'd hit this case and you'd just get a `Codec`, but it might be worth adding a typetest to be explicit that that's how it should work assuming that's correct.

Reply:
That's a good point, I'll add a test to make sure it's not type safe to do so.

Reply:
The conversation was [moved here](https://github.com/solana-labs/solana-web3.js/pull/1903#discussion_r1410943228).